build-bin/README.md

```diff
@@ -0,0 +1,203 @@
+
+The scripts here are portable. They do not include any CI provider-specific logic or ENV variables.
+This helps `.travis.yml` and `test.yml` (GitHub Actions) contain nearly the same contents, even if
+certain OpenZipkin projects need slight adjustments here. Portability has proven necessary, as
+OpenZipkin has had to transition CI providers many times due to feature and quota constraints.
+
+These scripts serve a second purpose, which is to facilitate manual releases, which has also
+happened many times due usually to service outages of CI providers. While tempting to use
+CI-provider specific tools, doing so can easily create a dependency where no one knows how to
+release anymore. Do not use provider-specific mechanisms to implement release flow. Instead,
+automate triggering of the scripts here.
+
+The only scripts that should be modified per project are in the base directory. Those in sub
+directories, such as [docker], should not vary project to project except accident of version drift.
+Intentional changes in sub directories should be relevant and tested on multiple projects to ensure
+they can be blindly copy/pasted.
+
+Conversely, the files in the base directory are project specific entry-points for test and deploy
+actions and are entirely appropriate to vary per project. Here's an overview:
+
+## Test
+
+Test builds and runs any tests of the project, including integration tests. CI providers should be
+configured to run tests on pull requests or pushes to the master branch, notably when the tag is
+blank. Tests should not run on documentation-only commits. Tests must not depend on authenticated
+resources, as running tests can leak credentials. Git checkouts should include the full history so
+that license headers or other git analysis can take place.
+
+ * [configure_test] - Sets up build environment for tests.
+ * [test] - Builds and runs tests for this project.
+
+### Example GitHub Actions setup
+
+A simplest GitHub Actions `test.yml` runs tests after configuring them, but only on relevant event
+conditions. The name `test.yml` and job `test` allows easy references to status badges and parity of
+the scripts it uses.
+
+The `on:` section obviates job creation and resource usage for irrelevant events. Notably, GitHub
+Actions includes the ability to skip documentation-only jobs.
+
+Combine [configure_test] and [test] into the same `run:` when `configure_test` primes file system
+cache.
+
+Here's a partial `test.yml` including only the aspects mentioned above.
+```yaml
+on:
+  push:
+    tags: ''
+    branches: master
+    paths-ignore: '**/*.md'
+  pull_request:
+    branches: master
+    paths-ignore: '**/*.md'
+
+jobs:
+  test:
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0  # full git history
+      - name: Test
+        run: build-bin/configure_test && build-bin/test
+```
+
+### Example Travis setup
```

Comment:
remove travis entirely so we don't need people to know about _both_ GA and travis? I saw your comment about the additional credits but it doesn't sound like it's worth bothering any longer.

Reply:
I can see this point for sure, mentioned in top-level issue comments by accident sorry! TL;DR; is that we can have a soft dep on travis in some cases where it helps. literally this file is the same for all the other repos, so it is nice to take advantage of commonality where used ex here https://github.com/openzipkin/docker-alpine/blob/master/.travis.yml

Reply:
Makes sense, thanks for the thorough reply!